src/main/java/practice/StreamPractice.java

```diff
@@ -74,7 +107,16 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * We want to reuse our validation in future, so let's write our own impl of Predicate
      * parametrized with Candidate in CandidateValidator.
      */
+
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream().filter(candidate -> candidate.getAge() >= 35
+                        && candidate.isAllowedToVote()
+                        && candidate.getNationality().equals("Ukrainian")
+                        && (Integer.parseInt(candidate.getPeriodsInUkr().split("-")[1])
+                        - Integer.parseInt(candidate.getPeriodsInUkr().split("-")[0]))
+                        >= 10)
```

Comment:
You have created CandidateValidator predicate. You can use it in your filter
```suggestion
        return candidates.stream()
        .filter(new CandidateValidator())
```